src/Drush/Generators/RecipeGenerator.php

```diff
@@ -4,78 +4,49 @@
-
-  /**
-   * The Drush generator command name.
-   *
-   * @var string
-   */
-  protected string $name = 'recipe';
-
-  /**
-   * The Drush generator command description.
-   *
-   * @var string
-   */
-  protected string $description = 'Generates a Recipe for adding new functionality to Drupal.';
-
-  /**
-   * The path to the templates for the generator.
-   *
-   * @var string
-   */
-  protected string $templatePath = __DIR__ . '/../../../templates';
+#[Generator(
+  name: 'recipe',
+  description: 'Generates a Recipe for adding new functionality to Drupal.',
+  aliases: ['rcp'],
+  templatePath: __DIR__ . '/../../../templates',
+  type: GeneratorType::OTHER,
+)]
+final class RecipeGenerator extends BaseGenerator {
 
   /**
    * {@inheritdoc}
    */
-  protected function getExtensionList(): array {
-    return [];
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  protected function generate(array &$vars): void {
-    $vars['recipe_name'] = $this->ask('What is the name of this recipe?', 'My Custom Recipe', '::validateRequired');
-    $vars['recipe_directory'] = $this->ask('In what directory should this recipe be saved under /recipes (ex. "my-recipe")?', 'my-custom-recipe', '::validateRequired');
-    $vars['recipe_type'] = $this->ask('What type of recipe is this (Site, Content Type, Workflow, etc)?', NULL, '::validateRequired');
-    $vars['recipe_description'] = $this->ask('What does this recipe do?', NULL, '::validateRequired');
+  protected function generate(array &$vars, AssetCollection $assets): void {
+    $ir = $this->createInterviewer($vars);
+    $vars['recipe_name'] = $ir->ask('What is the name of this recipe?', 'My Custom Recipe', new Required());
+    $vars['recipe_directory'] = $ir->ask('In what directory should this recipe be saved under /recipes (ex. "my-recipe")?', 'my-custom-recipe', new Required());
+    $vars['recipe_type'] = $ir->ask('What type of recipe is this (Site, Content Type, Workflow, etc)?', NULL, new Required());
+    $vars['recipe_description'] = $ir->ask('What does this recipe do?', NULL, new Required());
     $vars['composer'] = $this->collectComposerInfo($vars);
     $vars['modules'] = $this->collectModules($vars);
     $vars['config'] = $this->collectConfig($vars);
 
     if (!empty($vars['composer'])) {
-      $this->addFile('composer.json', 'composer/composer.json');
+      $assets->addFile('composer.json', '/composer/composer.json.twig');
     }
 
-    $this->addFile('recipe.yml', 'recipe/recipe.yml');
+    $assets->addFile('recipe.yml', '/recipe/recipe.yml.twig');
   }
 
   /**
    * Returns destination for generated recipes.
    *
    * This is a temporary workaround until Drupal generator supports Recipes.
    */
-  public function getDestination(array $vars): ?string {
-    return $this->drupalContext->getDrupalRoot() . '/recipes/custom/' . $vars['recipe_directory'];
+  public function getDestination(array $vars): string {
+    return DRUPAL_ROOT . '/recipes/custom/' . $vars['recipe_directory'];
```

Comment:
We recommends keeping recipes above the webroot now, and no custom/contrib subfolder.  So ../recipes/